### PR TITLE
User peak

### PR DESCRIPTION
**Pull request by licode**
Implement user peak, so users have more flexible control of escape peak or pileup peaks.

### CODE REVIEW

**Comment by tacaswell** [comment]
Can you explain the purpose of this a bit more?

**Comment by licode** [comment]
There are several reasons to add user peaks:
- User-peaks can be used as a quick replacement for a pileup peak. Pileup peaks are generated due to detector inefficiency. If users are confident that certain peak is not an element peak, but from detector defects, users can simply add user-peaks instead of spending long time working on pileup peaks.
- User-peaks can be used to represent escape peaks. 
- User-peaks can be used to represent other unknown peaks due to detector defects or scattering process.  
- This feature has been required by scientists at HXN for XRF fitting.

**Comment by licode** [comment]
anyone check on this? @ericdill @tacaswell ?

**Comment by ericdill** [comment]
Sure thing. I'll have a look shortly @licode

File: skbeam/core/fitting/xrf_model.py [hunk 1]
```diff
@@ -420,13 +420,19 @@ def add_param(self, kind, element, constraint=None):
                 if constraint is not None:
                     self._element_strategy[param_name] = constraint
                 self.params.update({param_name: new_pos})
-        else:
+        elif 'user' in element.lower():  # for user peak
```
Comment (ericdill) [inline skbeam/core/fitting/xrf_model.py]: Can you add a note about this in the docstring for this function?  It is not immediately evident from the current docstring that `add_param` supports the notion of a `user_peak`

**Comment by ericdill** [comment]
Are you using this in PyXRF?  I'm curious to see how this is being used in the wild. Or is the plan to get it in to scikit-beam and then update PyXRF to use it?

**Comment by ericdill** [comment]
Looks like you are missing coverage on all of the if blocks in the new `elif 'user' ...` block in xrf_model.py.  
The lines are 

    840, 851, 856, 862, 868

Can you make sure to hit those if blocks in the tests?

**Comment by ericdill** [comment]
Only a couple of minor comments regarding test coverage and adding documentation to the `add_param` function. Otherwise, looks good.

**Comment by licode** [comment]
Yes, this function was implemented at PyXRF already. Users can select the user peak from the scroll-down window the same as they choose element peaks.  I will add more coverage.

**Comment by ericdill** [comment]
Great, thanks for updating the docs. Looks good to me.

**Pull request merged**